File: shopy-api/Models/Category.cs
namespace shopy_api.Models;

public class Category : ISoftDeletable
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Slug { get; set; } = string.Empty;
    public string? ImageUrl { get; set; }

    public Guid? ParentCategoryId { get; set; }
    public Category? ParentCategory { get; set; }
    public ICollection<Category> ChildCategories { get; set; } = [];

    public ICollection<Product> Products { get; set; } = [];

    public DateTime CreatedAt { get; set; }
    public bool IsDeleted { get; set; }
}

File: shopy-seller/lib/models/catalog/category.dart
class Category {
  final String id;
  final String name;
  final String slug;
  final String? parentCategoryId;

  const Category({required this.id, required this.name, required this.slug, this.parentCategoryId});

  factory Category.fromJson(Map<String, dynamic> json) {
    return Category(
      id: json['id'] as String,
      name: json['name'] as String,
      slug: json['slug'] as String,
      parentCategoryId: json['parentCategoryId'] as String?,
    );
  }
}
